sync/sync_head.go

```diff
@@ -38,7 +38,11 @@ func (s *Syncer[H]) Head(ctx context.Context, _ ...header.HeadOption[H]) (H, err
 		return s.Head(ctx)
 	}
 	defer s.getter.Unlock()
-	netHead, err := s.getter.Head(ctx, header.WithTrustedHead[H](sbjHead))
+
+	reqCtx, cancel := context.WithTimeout(ctx, time.Second*2)
+	defer cancel()
+
+	netHead, err := s.getter.Head(reqCtx, header.WithTrustedHead[H](sbjHead))
```

Comment:
Ideally, we wouldn't wait until either the new header is rcvd from GossipSub or this request finishes. Right now we are waiting for the head to finish while we should just return a new gossiper header.

Reply:
Should land in an issue for a separate PR

Reply:
https://github.com/celestiaorg/go-header/issues/153